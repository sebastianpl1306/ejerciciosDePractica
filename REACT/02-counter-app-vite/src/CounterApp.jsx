import { useState } from 'react';
import PropTypes from "prop-types";

export const CounterApp =  ({ value }) => {

    //El useState recibe como parametro el valor de inicialización y este se desestructura en una variable llamada counter y una función que permite modificar
    //el valor de counter
    const [ counter, setCounter ] = useState( value );

    const handleAdd = () =>{
        // setCounter( counter + 1 );
        // setCounter(1000);
        setCounter((c)=> c + 1);
    }

    const handleSubstract = () => setCounter(counter-1);

    const handleReset = () => setCounter(value);

    return(
        <>
            <h1>counterApp</h1>
            <h2 data-testid="container-counter">{ counter }</h2>
            <button onClick={handleAdd}>+1</button>
            <button onClick={handleSubstract}>-1</button>
            <button aria-label="btn-reset" onClick={handleReset}>Reset</button>
        </>
    )
}

CounterApp.prototypes = {
    value: PropTypes.number
}