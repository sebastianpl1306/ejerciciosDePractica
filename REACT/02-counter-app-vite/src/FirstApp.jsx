import PropTypes from "prop-types";

const info = {
  name: 'Sebastian',
  lastname: 'Pabon Lopez'
};

const getName = (info) =>{
  return `Hello ${info.name} ${info.lastname} welcome!`;
}

export const FirstApp = ({title, number}) => {

  return (
    <>
      <h1>{ getName(info) }</h1>
      <p>FirstApp {title} {number+1}</p>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  number: PropTypes.number
};