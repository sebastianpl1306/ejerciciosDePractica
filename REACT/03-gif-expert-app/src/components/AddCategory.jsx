import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({onAddNewCategory}) => {
    const [inputValue, setinputValue] = useState('One Punch')

    const onInputChage = ({target}) =>{
        setinputValue(target.value);
    }

    const onSubmit = (event) =>{
        event.preventDefault();
        const value = inputValue.trim();
        if (value.length < 3 ) return;

        setinputValue('');
        onAddNewCategory(inputValue.trim());
    }

    return (
        <form onSubmit={ onSubmit } aria-label='form'>
            <input 
                type='text'
                placeholder='Buscar Gifs'
                value={ inputValue }
                onChange={ onInputChage }
            />
        </form>
    )
}

AddCategory.propTypes = {
    onAddNewCategory: PropTypes.func.isRequired
}