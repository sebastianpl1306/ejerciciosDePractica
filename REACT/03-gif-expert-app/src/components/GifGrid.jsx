import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";

export const GifGrid = ({ category }) => {
  const {gifs, isLoading} = useFetchGifs(category);

  return (
    <div>
        <h3>{category}</h3>
        {isLoading && (<h6>Cargando...</h6>)}
        <div className="card-grid">
          {
            gifs.map((images)=><GifItem key={images.id} {...images}/>)
          }
        </div>
    </div>
  )
}
