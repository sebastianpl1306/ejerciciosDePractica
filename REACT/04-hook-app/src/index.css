body{
    padding: 20px;
}

button{
    margin-right: 10px;
}

a{
    margin: 5px;
}