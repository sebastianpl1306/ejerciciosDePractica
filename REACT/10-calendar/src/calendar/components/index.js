export * from './CalendarEvent';
export * from './CalendarModal';
export * from './FabAddNew';
export * from './FabDelete';
export * from './Navbar';