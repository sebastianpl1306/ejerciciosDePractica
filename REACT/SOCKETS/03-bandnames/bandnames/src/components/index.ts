export * from './AddBand';
export * from './BandList';
export * from './BandChar';