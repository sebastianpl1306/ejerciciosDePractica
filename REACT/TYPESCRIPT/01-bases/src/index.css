html, body{
  background-color: #282c34;
  color: white;
  margin: 0;
  padding: 10px;
}