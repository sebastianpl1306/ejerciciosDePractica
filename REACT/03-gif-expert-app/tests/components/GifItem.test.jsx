import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components"

describe('Pruebas en <GifItem/>', () => {
    const title = 'One Punch';
    const url = 'https://media4.giphy.com/media/v1.Y2lkPTc5MGI3NjExY2Q5MGJkOGIyYTc2MmQ1YjUwNjRkMmE4ZjY1ZWFlZWZmNTI3MzczYSZjdD1n/yo3TC0yeHd53G/giphy.gif';

    test('Snapshopt del componente', () => {
        const { container } = render(<GifItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();
    })

    test('El src y el alt deben tener su url y titulo respectivo', () => {
        render(<GifItem title={title} url={url}/>);
        const {src, alt} = screen.getByRole('img');

        expect( src ).toBe( url );
        expect( alt ).toBe( title );
    })

    test('El texto del titulo debe estar presente en el componente', () => {
        render(<GifItem title={title} url={url}/>);
        expect(screen.getByText(title)).toBeTruthy();
    })
})