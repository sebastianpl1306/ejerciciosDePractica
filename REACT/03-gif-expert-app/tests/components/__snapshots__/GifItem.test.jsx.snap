// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> Snapshopt del componente 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch"
      src="https://media4.giphy.com/media/v1.Y2lkPTc5MGI3NjExY2Q5MGJkOGIyYTc2MmQ1YjUwNjRkMmE4ZjY1ZWFlZWZmNTI3MzczYSZjdD1n/yo3TC0yeHd53G/giphy.gif"
    />
    <p>
      One Punch
    </p>
  </div>
</div>
`;
